src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1525,68 +1534,461 @@ static void LoadRegPair(BYTE* p, int reg1, int reg2, UINT32 offset)
+    p += 4;
+
+    // label:
+    // arg
+    *(TADDR*)p = arg;
+    p += 8;
+    // target
+    *(TADDR*)p = target;
+    p += 8;
+
+    END_DYNAMIC_HELPER_EMIT();
 }
 
 PCODE DynamicHelpers::CreateReturn(LoaderAllocator * pAllocator)
 {
-    _ASSERTE(!"RISCV64: not implementation on riscv64!!!");
-    return NULL;
+    STANDARD_VM_CONTRACT;
+
+    BEGIN_DYNAMIC_HELPER_EMIT(4);
+
+    *(DWORD*)p = 0x00008067;// ret
+    p += 4;
+
+    END_DYNAMIC_HELPER_EMIT();
 }
 
 PCODE DynamicHelpers::CreateReturnConst(LoaderAllocator * pAllocator, TADDR arg)
 {
-    _ASSERTE(!"RISCV64: not implementation on riscv64!!!");
-    return NULL;
+    STANDARD_VM_CONTRACT;
+
+    BEGIN_DYNAMIC_HELPER_EMIT(24);
+
+    *(DWORD*)p = 0x00000297;// auipc t0, 0
+    p += 4;
+    *(DWORD*)p = 0x0102b503;// ld a0, 16(t0)
+    p += 4;
+    *(DWORD*)p = 0x00008067;// ret
+    p += 4;
+    *(DWORD*)p = 0x00000013;// nop, padding to make 8 byte aligned
+    p += 4;
+
+    // label:
+    // arg
+    *(TADDR*)p = arg;
+    p += 8;
+
+    END_DYNAMIC_HELPER_EMIT();
 }
 
 PCODE DynamicHelpers::CreateReturnIndirConst(LoaderAllocator * pAllocator, TADDR arg, INT8 offset)
 {
-    _ASSERTE(!"RISCV64: not implementation on riscv64!!!");
-    return NULL;
+    STANDARD_VM_CONTRACT;
+
+    BEGIN_DYNAMIC_HELPER_EMIT(32);
+
+    *(DWORD*)p = 0x00000297;// auipc t0, 0
+    p += 4;
+    *(DWORD*)p = 0x0182b503;// ld a0,24(t0)
+    p += 4;
+    *(DWORD*)p = 0x00053503;// ld a0,0(a0)
+    p += 4;
+    *(DWORD*)p = 0x00050513 | ((offset & 0xfff)<<20);// addi a0,a0,offset
+    p += 4;
+    *(DWORD*)p = 0x00008067;// ret
+    p += 4;
+    *(DWORD*)p = 0x00000013;// nop, padding to make 8 byte aligned
+    p += 4;
+
+    // label:
+    // arg
+    *(TADDR*)p = arg;
+    p += 8;
+
+    END_DYNAMIC_HELPER_EMIT();
 }
 
 PCODE DynamicHelpers::CreateHelperWithTwoArgs(LoaderAllocator * pAllocator, TADDR arg, PCODE target)
 {
-    _ASSERTE(!"RISCV64: not implementation on riscv64!!!");
-    return NULL;
+    STANDARD_VM_CONTRACT;
+
+    BEGIN_DYNAMIC_HELPER_EMIT(32);
+
+    *(DWORD*)p = 0x00000297;// auipc t0, 0
+    p += 4;
+    *(DWORD*)p = 0x0102b603;// ld a2,16(t0)
+    p += 4;
+    *(DWORD*)p = 0x0182b283;// ld t0,24(t0)
+    p += 4;
+    *(DWORD*)p = 0x00028067;// jr t0
+    p += 4;
+
+    // label:
+    // arg
+    *(TADDR*)p = arg;
+    p += 8;
+
+    // target
+    *(TADDR*)p = target;
+    p += 8;
+    END_DYNAMIC_HELPER_EMIT();
 }
 
 PCODE DynamicHelpers::CreateHelperWithTwoArgs(LoaderAllocator * pAllocator, TADDR arg, TADDR arg2, PCODE target)
 {
-    _ASSERTE(!"RISCV64: not implementation on riscv64!!!");
-    return NULL;
+    STANDARD_VM_CONTRACT;
+
+    BEGIN_DYNAMIC_HELPER_EMIT(48);
+
+    *(DWORD*)p = 0x00000297;// auipc t0, 0
+    p += 4;
+    *(DWORD*)p = 0x0182b603;// ld a2,24(t0)
+    p += 4;
+    *(DWORD*)p = 0x0202b683;// ld a3,32(t0)
+    p += 4;
+    *(DWORD*)p = 0x0282b283;// ld t0,40(t0)
+    p += 4;
+    *(DWORD*)p = 0x00028067;// jr t0
+    p += 4;
+
```

Comment:
Seems padding `nop` should also be here.

Reply:
Done